scale-tests/run_spark_scale_test.sh

```diff
@@ -0,0 +1,226 @@
+  exit 1
+fi
+
+
+## Get a stable shell 
+git clone git@github.com:mesosphere/spark-build.git
+cd spark-build
+docker build -t mesosphere/dcos-commons:${LOGNAME}-spark scale-tests/
+
+
+## Script to be executed inside the shell
+###### cat > /tmp/spark_scale_script.sh << EOF
+## Setup cluster
+
+dcos cluster setup --insecure --username=${CLUSTER_USERNAME} --password=${CLUSTER_PASSWORD} ${CLUSTER_URL}
+eval "$(ssh-agent)"
+ssh-add -k /ssh/key
+
+## Add Spark stub
+dcos package repo add --index=0 spark-aws https://universe-converter.mesosphere.com/transform?url=https://infinity-artifacts.s3.amazonaws.com/permanent/spark/assets/scale-testing/stub-universe-spark.json
+
+## Deploy dispatchers
+./scale-tests/deploy-dispatchers.py \
+  --quota-drivers-cpus $QUOTA_DRIVERS_CPUS \
+  --quota-drivers-mem $QUOTA_DRIVERS_MEM \
+  --quota-executors-cpus $QUOTA_EXECUTORS_CPUS \
+  --quota-executors-mem $QUOTA_EXECUTORS_MEM \
+  $NUM_DISPATCHERS $TEST_NAME ${TEST_NAME}-${NUM_DISPATCHERS}
+
+## Deploy GPU dispatchers
+
+## There is a bug in the script that auto-fails if a dispatcher already exists so
+## we uninstall the first n dispatchers to update them for GPU ones
+for ((idx=0; i<${NUM_GPU_DISPATCHERS}; i++))
+do
+  dcos package uninstall spark --app-id=/${TEST_NAME}spark-0${idx} --yes
+done
+
+./scale-tests/deploy-dispatchers.py \
+  --quota-drivers-cpus $QUOTA_DRIVERS_CPUS \
+  --quota-drivers-mem $QUOTA_DRIVERS_MEM \
+  --quota-executors-cpus $QUOTA_EXECUTORS_CPUS \
+  --quota-executors-mem $QUOTA_EXECUTORS_MEM \
+  --quota-executors-gpus $QUOTA_EXECUTORS_GPUS \
+  $NUM_GPU_DISPATCHERS $TEST_NAME ${TEST_NAME}-${NUM_GPU_DISPATCHERS}
+
+DISPATCHERS_JSON_FILE="${TEST_NAME}-${NUM_DISPATCHERS}-dispatchers.json"
+
+## Upload dispatchers file to S3
+aws s3 cp --acl public-read \
+  "${DISPATCHERS_JSON_FILE}" \
+  "s3://${TEST_S3_BUCKET}/${TEST_S3_FOLDER}/${DISPATCHERS_JSON_FILE}"
+
+## Deploy streaming infrastructure
+SERVICE_NAMES_PREFIX="${TEST_NAME}"
+INFRASTRUCTURE_JSON_FILE="${TEST_NAME}-infrastructure.json"
+KAFKA_ZK_CONFIG=${KAFKA_ZK_CONFIG:-scale-tests/configs/kafka-zookeeper-options.json}
+KAFKA_CONFIG=${KAFKA_CONFIG:-scale-tests/configs/kafka-options.json}
+CASSANDRA_CONFIG=s${CASSANDRA_CONFIG:-cale-tests/configs/cassandra-options.json}
+
+./scale-tests/setup_streaming.py $INFRASTRUCTURE_JSON_FILE \
+  --service-names-prefix $SERVICE_NAMES_PREFIX \
+  --kafka-zookeeper-config $KAFKA_ZK_CONFIG \
+  --kafka-cluster-count $NUM_KAFKA_CLUSTERS \
+  --kafka-config $KAFKA_CONFIG \
+  --cassandra-cluster-count $NUM_CASSANDRA_CLUSTERS \
+  --cassandra-config $CASSANDRA_CONFIG
+
+
+## Upload infrastructure file to S3
+aws s3 cp --acl public-read \
+  "${INFRASTRUVTURE_JSON_FILE}" \
+  "s3://${TEST_S3_BUCKET}/${TEST_S3_FOLDER}/${INFRASTRUCTURE_JSON_FILE}"
+
+## Prepare dispatcher list files
+NUM_INFINITE_DISPATCHERS="$(expr "${NUM_DISPATCHERS}" - "${NUM_FINITE_DISPATCHERS}")"
+DISPATCHERS_FINITE_JSON_FILE="${TEST_NAME}-finite-dispatchers.json"
```

Comment:
Rather launch two different sets of dispatchers than having to manipulate the list of dispatchers.